pkg/chunk/schema_config.go

```diff
@@ -69,116 +69,27 @@ func (d *DayTime) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	// hour.  Only the day matters, not the time within the day.
-	DailyBucketsFrom      flagext.DayValue
-	Base64ValuesFrom      flagext.DayValue
-	V4SchemaFrom          flagext.DayValue
-	V5SchemaFrom          flagext.DayValue
-	V6SchemaFrom          flagext.DayValue
-	V9SchemaFrom          flagext.DayValue
-	BigtableColumnKeyFrom flagext.DayValue
-
-	// Config for the index & chunk tables.
-	OriginalTableName string
-	UsePeriodicTables bool
-	IndexTablesFrom   flagext.DayValue
-	IndexTables       PeriodicTableConfig
-	ChunkTablesFrom   flagext.DayValue
-	ChunkTables       PeriodicTableConfig
+	fileName       string
+	legacyFileName string
 }
 
 // RegisterFlags adds the flags required to config this to the given FlagSet.
 func (cfg *SchemaConfig) RegisterFlags(f *flag.FlagSet) {
-	flag.StringVar(&cfg.fileName, "config-yaml", "", "Schema config yaml")
-	cfg.legacy.RegisterFlags(f)
-}
-
-// RegisterFlags adds the flags required to config this to the given FlagSet.
-func (cfg *LegacySchemaConfig) RegisterFlags(f *flag.FlagSet) {
-	flag.StringVar(&cfg.StorageClient, "chunk.storage-client", "aws", "Which storage client to use (aws, gcp, cassandra, inmemory).")
-	f.Var(&cfg.DailyBucketsFrom, "dynamodb.daily-buckets-from", "The date (in the format YYYY-MM-DD) of the first day for which DynamoDB index buckets should be day-sized vs. hour-sized.")
-	f.Var(&cfg.Base64ValuesFrom, "dynamodb.base64-buckets-from", "The date (in the format YYYY-MM-DD) after which we will stop querying to non-base64 encoded values.")
-	f.Var(&cfg.V4SchemaFrom, "dynamodb.v4-schema-from", "The date (in the format YYYY-MM-DD) after which we enable v4 schema.")
-	f.Var(&cfg.V5SchemaFrom, "dynamodb.v5-schema-from", "The date (in the format YYYY-MM-DD) after which we enable v5 schema.")
-	f.Var(&cfg.V6SchemaFrom, "dynamodb.v6-schema-from", "The date (in the format YYYY-MM-DD) after which we enable v6 schema.")
-	f.Var(&cfg.V9SchemaFrom, "dynamodb.v9-schema-from", "The date (in the format YYYY-MM-DD) after which we enable v9 schema (Series indexing).")
-	f.Var(&cfg.BigtableColumnKeyFrom, "bigtable.column-key-from", "The date (in the format YYYY-MM-DD) after which we use bigtable column keys.")
-
-	f.StringVar(&cfg.OriginalTableName, "dynamodb.original-table-name", "cortex", "The name of the DynamoDB table used before versioned schemas were introduced.")
-	f.BoolVar(&cfg.UsePeriodicTables, "dynamodb.use-periodic-tables", false, "Should we use periodic tables.")
-
-	f.Var(&cfg.IndexTablesFrom, "dynamodb.periodic-table.from", "Date after which to use periodic tables.")
-	cfg.IndexTables.RegisterFlags("dynamodb.periodic-table", "cortex_", f)
-	f.Var(&cfg.ChunkTablesFrom, "dynamodb.chunk-table.from", "Date after which to write chunks to DynamoDB.")
-	cfg.ChunkTables.RegisterFlags("dynamodb.chunk-table", "cortex_chunks_", f)
+	flag.StringVar(&cfg.fileName, "schema-config-file", "", "The path to the schema config file.")
+	// TODO(gouthamve): Add a metric and log line for this.
```

Comment:
Not sure the metric is that important, but we could already add a log line. We can do it in the `loadFromFile()`.

Reply:
I agree that a log line is appropriate, to note the use of a deprecated flag. I don't think a metric is useful.